Chart/assets/js/utils.js

```diff
@@ -0,0 +1,173 @@
+//functionality
+let input1;
+let input2;
+let start_date;
+let end_date;
+let interval;
+let period;
+let average;
+let percent;
+let rate;
+let sum;
+const average_div=document.getElementById("average");
+const rate_div=document.getElementById("rate");
+const percent_div=document.getElementById("percent");
+
+
+function formatDate(date) {
+    const year = date.getFullYear();
+    const month = String(date.getUTCMonth() + 1).padStart(2, '0'); // Months are zero-based
+    const day = String(date.getUTCDate()).padStart(2, '0');
+    const hours = String(date.getUTCHours()).padStart(2, '0');
+    const minutes = String(date.getUTCMinutes()).padStart(2, '0');
+
+    return `${year}-${month}-${day}-${hours}:${minutes}`;
+}   
+let now1 = new Date();
+const formattedDate = formatDate(now1);
+console.log(formattedDate);
+
+let quarter = new Date(now1);
+quarter.setMinutes(now1.getMinutes()-15);
+console.log(formatDate(quarter));
+
+
+const dropdownItems1 = document.querySelectorAll('.currency1');
+const firstinput = document.getElementById('currency1');
+dropdownItems1.forEach(item => {
+    item.addEventListener('click', function() {
+        firstinput.textContent = this.textContent;
+        document.getElementById("button1").textContent=this.textContent;
+        input1=this.textContent;
+        console.log(input1);
+        let flag1=document.getElementById("flag1")
+        flag(input1,flag1);
+
+    });
+});
+const dropdownItems2 = document.querySelectorAll('.currency2');
+const secondinput = document.getElementById('currency2');
+dropdownItems2.forEach(item => {
+    item.addEventListener('click', function() {
+
+        secondinput.textContent = this.textContent; 
+        document.getElementById("button2").textContent=this.textContent;
+        input2=this.textContent;
+        let flag2=document.getElementById("flag2")
+        flag(input2,flag2);
+        Exchange = input1+input2;
+        console.log(Exchange);
+
+});
+
+});
+const button1 = document.getElementById('quarter');
+const button2 = document.getElementById('hour');
+const button3 = document.getElementById('day');
+const button4 = document.getElementById('week');
+const button5 = document.getElementById('month');
+button1.addEventListener('click', function() {
```

Comment:
if clicked before fetching data getting unhandled error , choose whether to fething data and setting it at first rendering , or disable these buttons if currancies not selected yet